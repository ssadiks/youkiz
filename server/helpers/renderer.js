import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import serialize from 'serialize-javascript';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Routes from '../../src/Routes';
import HeaderContainer from '../../src/components/container/HeaderContainer/HeaderContainer';

const renderer = (req, store) => {
  const muiTheme = getMuiTheme({
    userAgent: req.headers['user-agent']
  });

  const content = renderToString(
    <Provider store={store}>
      <MuiThemeProvider muiTheme={muiTheme}>
        <StaticRouter location={req.path} context={{}}>
          <div>
            {renderRoutes(Routes)}
          </div>
        </StaticRouter>
      </MuiThemeProvider>
    </Provider>
  );

  return `
    <html>
      <head>
        <link href="css/style.css" rel="stylesheet">
      </head>
      <body>
        <div id="root">${content}</div>
        <script>
          window.INITIAL_STATE = ${serialize(store.getState())}
        </script>
        <script src="js/vendors.js"></script>
        <script src="js/bundle.js"></script>
      </body>
    </html>
  `;
};

export default renderer;
