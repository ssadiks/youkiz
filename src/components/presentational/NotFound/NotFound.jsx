import React from 'react';

export const NotFound = () => (
  <div>
    <span>NOT FOUND</span>
  </div>
);
