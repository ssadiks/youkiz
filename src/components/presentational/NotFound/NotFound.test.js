import React from 'react';
import renderer from 'react-test-renderer';
import NotFound from './NotFound';

describe('NotFound render', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <NotFound />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
