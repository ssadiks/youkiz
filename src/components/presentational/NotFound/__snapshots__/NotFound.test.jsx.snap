// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound render renders correctly 1`] = `
<div>
  <span>
    NOT FOUND
  </span>
</div>
`;
